software/tracksight/frontend/src/app/components/graph.tsx

```diff
@@ -2,44 +2,112 @@
 
 import { useState, useEffect, React } from 'react';
 import Plot from 'react-plotly.js';
-
+import DropdownMenu from './dropdown_menu.tsx';
+import { Button } from 'antd';
 import WebSocketComponent from './web_socket.tsx';
 
-
 const Graph = (props) => {
-	const [data, setData] = useState({});
-    const [formattedData, setFormattedData] = useState({
-        x: [],
-        y: [],
-        type: 'scatter',
-        mode: 'lines+markers',
-        marker: {color: 'red'},
-    });
+    
+    const [data, setData] = useState({});
+    const [formattedData, setFormattedData] = useState([]);
+    const [graphName, setGraphName] = useState("Empty");
+    const [signals, setSignals] = useState([])
```

Comment:
We spoke about having `graph_id`s to keep track of the one socket multiple plots, but I think that can come in a separate PR? Would be good to have this merged in for us to work off